### PR TITLE
Refactoring

### PR DESCRIPTION
**Pull request by koonwen**

### CODE REVIEW

File: lib/batcher.mli [hunk 1]
```diff
@@ -1,34 +1,93 @@
...
+  val init : unit -> 'a t
+  (** [init ()] returns a new instance of the data structure. *)
+
+  val run : 'a t -> Task.pool -> 'a wrapped_op array -> unit
+  (** [run t pool ops num] when called with a data structure ['a t], and
+      a thread pool [pool], executes all the operations in [ops],
+      possibly using parallelism to improve the speed of the
+      operation. *)
+
+end
```
Comment (koonwen) [inline lib/batcher.mli:43]: Why did you make the type of the BatchedDS polymorphic? I thought we agreed to keep them abstract and provide the DS's as functors

File: data/btree.ml [hunk 2]
```diff
@@ -1,3 +1,329 @@
...
+      mutable keys: V.t Finite_vector.t;              (* keys themselves *)
+      mutable values: 'a Finite_vector.t;              (* values *)
+      leaf: bool;
+      mutable children: 'a node Finite_vector.t;
+      mutable no_elements: int;             (* number of elements in the node and subtrees  *)
+    }
+
+    type 'a t = {
+      mutable root: 'a node;
+      max_children: int;
```
Comment (koonwen) [inline data/btree.ml:20]: `min_children` like we discussed?

Comment (kiranandcode) [reply data/btree.ml:20]: Yes, should be min children

File: lib/batcher.mli (hunk 1 above)
Comment (kiranandcode) [reply lib/batcher.mli:43]: After starting to refactor your experiments code, I realised that you were right, and my  initial guess was wrong - it really does make sense to have a separate polymorphic interface -- hence the S1 and Make1 functors.

**Pull request merged**